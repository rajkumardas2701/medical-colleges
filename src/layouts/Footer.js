import '../styles/Footer.css';

const getYear = () => {
  const d = new Date();
  return d.getFullYear();
};

const Footer = () => (
  <footer>
    <p>
      {'Â© '}
      {
       getYear()
      }
      {' '}
      |
      {' '}
      <b>Rajkumar Das</b>
      {' '}
      All rights reserved
    </p>
  </footer>
);

export default Footer;
