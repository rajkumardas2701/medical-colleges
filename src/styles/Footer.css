footer {
  position: fixed;
  left: 0;
  bottom: 0;
  width: 100%;
  background-color: rgba(41, 29, 29, 0.856);
  color: white;
  text-align: center;
  height: 50px;
}

footer > p {
  font-family:Arial, Helvetica, sans-serif;
  font-size: 18px;
}