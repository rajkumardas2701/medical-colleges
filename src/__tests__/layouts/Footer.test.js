import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Footer from '../../layouts/Footer';

afterEach(cleanup);

it('renders Footer component correctly', () => {
  const footer = renderer.create(<Footer />).toJSON();
  expect(footer).toMatchSnapshot();
});

it('should display developer name', () => {
  render(<Footer />);
  const devName = screen.getByText(/All rights reserved/);
  expect(devName).toBeInTheDocument();
});
