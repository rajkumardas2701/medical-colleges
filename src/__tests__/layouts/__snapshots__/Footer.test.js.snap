// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer component correctly 1`] = `
<footer>
  <p>
    Â© 
    2021
     
    |
     
    <b>
      Rajkumar Das
    </b>
     
    All rights reserved
  </p>
</footer>
`;
